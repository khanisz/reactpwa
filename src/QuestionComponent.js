import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import TextFields from "./TextFields";

const styles = {
  card: {
    minWidth: 275,
    margin: 10
  },
  bullet: {
    display: "inline-block",
    margin: "0 2px",
    transform: "scale(0.8)"
  },
  title: {
    fontSize: 14
  },
  pos: {
    marginBottom: 12
  },
  container: {
    display: "flex",
    flexWrap: "wrap"
  },
  textField: {
    marginLeft: 10,
    marginRight: 10
  },
  dense: {
    marginTop: 16
  },
  menu: {
    width: 200
  }
};

function QuestionComponent(props) {
  const { classes } = props;

  const bull = <span className={classes.bullet}>â€¢</span>;
  const { text } = props;
  var value = "";
  function handleSubmit(event) {
    alert("A name was submitted: " + value);
    event.preventDefault();
  }
  function handleChange(event) {
    value = event.target.value;
  }

  const { title, question, answer, resolution } = props;

  return (
    <Card className={classes.card}>
      <CardContent>
        <Typography
          className={classes.title}
          color="textSecondary"
          gutterBottom
        />
        <Typography variant="h5" component="h2">
          {title}
        </Typography>
        <Typography className={classes.pos} color="textSecondary" />
        <Typography component="p">{question}</Typography>
        <TextFields answer={answer} resolution={resolution} />
      </CardContent>
    </Card>
  );
}

QuestionComponent.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(QuestionComponent);
