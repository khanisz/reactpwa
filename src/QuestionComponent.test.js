import React from "react";
import { shallow } from "enzyme";
import QuestionComponent from "./QuestionComponent";

describe("TextComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<QuestionComponent debug />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<QuestionComponent />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with answer", () => {
    const answer = "Tytul";
    const component = shallow(<QuestionComponent answer={answer} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with resolution", () => {
    const resolution = "Lore ipsum";
    const component = shallow(<QuestionComponent resolution={resolution} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with question, answer", () => {
    const question = 'Pytanie';
    const answer = "Tytul";
    const component = shallow(
      <QuestionComponent question = {question} answer={answer}  />
    );
    expect(component).toMatchSnapshot();
  });


  it("should render correctly with question, answer, resolution", () => {
    const question = "Pytanie";
    const answer = "Tytul";
    const resolution = "Lore ipsum";
    const component = shallow(
      <QuestionComponent
        question={question}
        answer={answer}
        resolution={resolution}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
