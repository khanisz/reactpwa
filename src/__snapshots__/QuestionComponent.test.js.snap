// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextComponent should render correctly in "debug" mode 1`] = `
<QuestionComponent
  classes={
    Object {
      "bullet": "QuestionComponent-bullet-2",
      "card": "QuestionComponent-card-1",
      "container": "QuestionComponent-container-5",
      "dense": "QuestionComponent-dense-7",
      "menu": "QuestionComponent-menu-8",
      "pos": "QuestionComponent-pos-4",
      "textField": "QuestionComponent-textField-6",
      "title": "QuestionComponent-title-3",
    }
  }
  debug={true}
/>
`;

exports[`TextComponent should render correctly with answer 1`] = `
<QuestionComponent
  answer="Tytul"
  classes={
    Object {
      "bullet": "QuestionComponent-bullet-2",
      "card": "QuestionComponent-card-1",
      "container": "QuestionComponent-container-5",
      "dense": "QuestionComponent-dense-7",
      "menu": "QuestionComponent-menu-8",
      "pos": "QuestionComponent-pos-4",
      "textField": "QuestionComponent-textField-6",
      "title": "QuestionComponent-title-3",
    }
  }
/>
`;

exports[`TextComponent should render correctly with no props 1`] = `
<QuestionComponent
  classes={
    Object {
      "bullet": "QuestionComponent-bullet-2",
      "card": "QuestionComponent-card-1",
      "container": "QuestionComponent-container-5",
      "dense": "QuestionComponent-dense-7",
      "menu": "QuestionComponent-menu-8",
      "pos": "QuestionComponent-pos-4",
      "textField": "QuestionComponent-textField-6",
      "title": "QuestionComponent-title-3",
    }
  }
/>
`;

exports[`TextComponent should render correctly with question, answer 1`] = `
<QuestionComponent
  answer="Tytul"
  classes={
    Object {
      "bullet": "QuestionComponent-bullet-2",
      "card": "QuestionComponent-card-1",
      "container": "QuestionComponent-container-5",
      "dense": "QuestionComponent-dense-7",
      "menu": "QuestionComponent-menu-8",
      "pos": "QuestionComponent-pos-4",
      "textField": "QuestionComponent-textField-6",
      "title": "QuestionComponent-title-3",
    }
  }
  question="Pytanie"
/>
`;

exports[`TextComponent should render correctly with question, answer, resolution 1`] = `
<QuestionComponent
  answer="Tytul"
  classes={
    Object {
      "bullet": "QuestionComponent-bullet-2",
      "card": "QuestionComponent-card-1",
      "container": "QuestionComponent-container-5",
      "dense": "QuestionComponent-dense-7",
      "menu": "QuestionComponent-menu-8",
      "pos": "QuestionComponent-pos-4",
      "textField": "QuestionComponent-textField-6",
      "title": "QuestionComponent-title-3",
    }
  }
  question="Pytanie"
  resolution="Lore ipsum"
/>
`;

exports[`TextComponent should render correctly with resolution 1`] = `
<QuestionComponent
  classes={
    Object {
      "bullet": "QuestionComponent-bullet-2",
      "card": "QuestionComponent-card-1",
      "container": "QuestionComponent-container-5",
      "dense": "QuestionComponent-dense-7",
      "menu": "QuestionComponent-menu-8",
      "pos": "QuestionComponent-pos-4",
      "textField": "QuestionComponent-textField-6",
      "title": "QuestionComponent-title-3",
    }
  }
  resolution="Lore ipsum"
/>
`;
